### PR TITLE
Cppcheck - Ignore build dir in invocation

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Ignore build dir in cppcheck invocation

This should stop cppcheck completely checking the dir.
Currently it checks the files, then matches the suppression and discards
the result

**To test:**
- Check cppcheck job runs

Noticed in #33028 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/inc/MantidKernel/Strings.h [hunk 1]
```diff
@@ -360,11 +360,9 @@ template <typename Integer> std::vector<std::vector<Integer>> parseGroups(const
     const auto tokens = Kernel::StringTokenizer(
         str, "+", Kernel::StringTokenizer::TOK_TRIM | Kernel::StringTokenizer::TOK_IGNORE_EMPTY);
     std::vector<Integer> group;
-    group.reserve(tokens.count());
-    for (const auto &t : tokens) {
-      // add this number to the group we're about to add
-      group.emplace_back(boost::lexical_cast<Integer>(t));
-    }
+    group.reserve(tokens.count() + 1);
```
Comment (martyngigg) [inline Framework/Kernel/inc/MantidKernel/Strings.h:363]: Could you help me understand the `+1`?

Comment (DavidFair) [reply Framework/Kernel/inc/MantidKernel/Strings.h:363]: I noticed we added n `tokens` then emplaced back `group` straight after for n+1
Do you think it's worth a comment in this case? I was on the fence about it

Comment (martyngigg) [reply Framework/Kernel/inc/MantidKernel/Strings.h:363]: Is the `emplace_back` not into `groups` (plural) and not `group`. Admitedley that's hard to parse too...

Comment (DavidFair) [reply Framework/Kernel/inc/MantidKernel/Strings.h:363]: Good point, I'll rename the variables as it's subtle :+1:

File: Framework/Kernel/inc/MantidKernel/Strings.h [hunk 2]
```diff
@@ -359,13 +359,11 @@ template <typename Integer> std::vector<std::vector<Integer>> parseGroups(const
...
     const auto tokens = Kernel::StringTokenizer(
         str, "+", Kernel::StringTokenizer::TOK_TRIM | Kernel::StringTokenizer::TOK_IGNORE_EMPTY);
-    std::vector<Integer> group;
-    group.reserve(tokens.count());
-    for (const auto &t : tokens) {
-      // add this number to the group we're about to add
-      group.emplace_back(boost::lexical_cast<Integer>(t));
-    }
-    groups.emplace_back(std::move(group));
+    std::vector<Integer> currentGroup;
```
Comment (martyngigg) [inline Framework/Kernel/inc/MantidKernel/Strings.h:362]: Much better name. Thanks!

**Review by martyngigg — APPROVED** [review]
It's good that we can reduce the work cppcheck does. 👍🏻
